ui/views.py

```diff
@@ -190,34 +194,31 @@ def create_repo(request):
     )
 
 
-@login_required
-def listing(request, repo_slug):
-    """
-    View available LearningResources by repository.
-    """
-    # Enforce repository access restrictions.
-
-    # may work better as a database query, not likely a bottleneck though
-    repos = get_repos(request.user.id)
-    if repo_slug not in set([x.slug for x in repos]):
-        return HttpResponseForbidden("unauthorized")
-    repo = [x for x in repos if x.slug == repo_slug][0]
-    page = int(request.GET.get("page", "1"))
-    context = {
-        "repo_id": repo.id,
-        "repo_slug": repo.slug,
-        "repo": repo,
-        "courses": get_repo_courses(repo.id),
-        "runs": get_runs(repo.id),
-        "tags": get_user_tags(repo.id),
-        "resources": Paginator(
-            get_resources(repo.id), 20).page(page)
-    }
-    return render(
-        request,
-        "listing.html",
-        context,
-    )
+class RepositoryView(SearchView):
```

Comment:
This view, a subclass of a Haystack class-based view, is the meat of this PR. Mostly, it encapsulates search and pagination. Most of the code here is to heal with the slug for user permissions.